v1.5 final release: quick hits (#3249)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/contract.md

When the `contract` configuration is enforced, dbt will ensure that your model's returned dataset exactly matches the attributes you have defined in yaml:
- `name` and `data_type` for every column

> - Additional [`constraints`](resource-properties/constraints), as supported for this materialization + data platform

Reviewer: ```suggestion
- Additional [`constraints`](resource-properties/constraints), as supported for this materialization and data platform
```

> This is to ensure that downstream queriers of your model—inside and outside dbt—have a predictable and consistent set of columns to use in their analyses. Even a subtle change in data type, such as from `boolean` (`true`/`false`) to `integer` (`0`/`1`), could cause queries to fail in surprising ways.

Reviewer 2: > inside and outside dbt

This is money 🎯

> This is to ensure that downstream queriers of your model—inside and outside dbt—have a predictable and consistent set of columns to use in their analyses. Even a subtle change in data type, such as from `boolean` (`true`/`false`) to `integer` (`0`/`1`), could cause queries to fail in surprising ways.

Reviewer: ```suggestion
This is to ensure that downstream queries of your model (both within and outside dbt) have a predictable and consistent set of columns to use in their analyses. Even a subtle change in data type, such as from `boolean` (`true`/`false`) to `integer` (`0`/`1`), could cause queries to fail in surprising ways.
```

Author: I'm going to switch `downstream queriers` &rarr; `people querying your model downstream`. I do think it's important that we're talking about people here, and not just impersonal queries/systems.

> both within and outside dbt

Agree with "both." I'm going to keep this parenthetical inside m-dashes for emphasis, rather than just an aside, because I think it's a really important point. Doug seems to agree :) https://github.com/dbt-labs/docs.getdbt.com/pull/3249#discussion_r1175370722

The `data_type` defined in your yaml file must match a data type your data platform recognizes. dbt does not do any type aliasing itself. If your data platform recognizes both `int` and `integer` as corresponding to the same type, then they will return a match.

> That said, dbt will **not** include size, precision, or scale in its data type comparison. We don't think you should sweat the difference between `varchar(256)` and `varchar(257)`, because it almost never affects the experience of downstream queriers. If you need a more-precise comparison, it's possible to accomplish with a custom test.

Reviewer 2: > dbt will **not** include size, precision, or scale in its data type comparison

This is a key insight! 👀

Author: > If you need a more-precise comparison, it's possible to accomplish with a custom test.

I was being a touch lazy here, but I think the point stands! I'll link to https://docs.getdbt.com/guides/best-practices/writing-custom-generic-tests

> That said, dbt will **not** include size, precision, or scale in its data type comparison. We don't think you should sweat the difference between `varchar(256)` and `varchar(257)`, because it almost never affects the experience of downstream queriers. If you need a more-precise comparison, it's possible to accomplish with a custom test.

Reviewer: ```suggestion
That said, when comparing data types in dbt, it doesn't take into account certain details such as size, precision, or scale in its data type comparison. The difference between `varchar(256)` and `varchar(257)` isn't significant enough to affect the experience of downstream queriers. If you need a more precise comparison, it's possible to accomplish it with a custom test.

```

Author: Could I have special dispensation to keep this one a bit more personal/informal in tone? This is our opinion as practitioners (&rarr; dbt maintainers), not an absolute truth

Personal/informal:
> We don't think you should sweat the difference between `varchar(256)` and `varchar(257)`, because it almost never affects the experience of downstream queriers.

Impersonal/formal:
> The difference between `varchar(256)` and `varchar(257)` is really not significant enough to affect the experience of downstream queriers.

Reviewer: for sure no worries and i love informal tones!! i was only thinking of the word 'sweat' with regard to future translation to be honest so that's why i tried prepping it.

Author: totally fair! it'll be a fun bridge to cross when we get there

> on pense que la difference entre `varchar(256)` et `varchar(257)` n'est pas très grave, t'inquiète minot

Reviewer: haha oui oui, c'est pas grave!
